Chase_step2_OutcomeDatesAssignedMacro.sas

```diff
@@ -1,143 +1,195 @@
+to the pregnancy outcome groups (i.e., Step 2 of the cohort derivation).
+
+Modifications:
+	- 04-29-24: CDL made comments and applied standardized set up.
+
+*******************************************************************************/
 
 
 %macro outdts(gap);
-/**/
+
+/*Testing*/
 /*%let gap=7;*/
 /*%let i=1;*/
 
- proc sql stimer;  
-  %do i=1 %to 4;
+ 	proc sql stimer;  
 
-  
-      create table getdate0_&gap._&i as
-       select distinct a.patient_deid,dt_outcomegroup_start,dt_outcomegroup_end, outcomegrp, 
-              a.outcome_assigned&i.,outcome_assigned_codetype&i., outcome_class_assigned&i, 
+		%do i=1 %to 4; /*Do this once for each algorithm*/
+
+		/*Create an initial dataset with the pregnancy outcome dates*/
+      	create table getdate0_&gap._&i as
+       	select distinct a.patient_deid, dt_outcomegroup_start, dt_outcomegroup_end, outcomegrp, 
+              a.outcome_assigned&i., outcome_assigned_codetype&i., outcome_class_assigned&i, /*Maybe dont need the a. for outcome_assigned?*/
               min(case
 
-              /*delivery assigned - pick encounter inpt proc [DelivPR_Inp], then any proc (conc pair), then diag-cd(pair)*/
+					   /**DELIVERY outcome dates**/
+
+              		   /*delivery assigned - pick encounter inpt proc [DelivPR_Inp], then any proc (conc pair), then diag-cd(pair)*/
+			  		   /*If inpatient delivery w procedure code, then outcome date is the encounter date*/
                        when outcome_class_assigned&i ='Delivery'  and DelivPR_Inp=1 then Enc_Date
 
+					   /*Grab the first concordant procedure code -- deliveries (order doesnt matter from abortions since based on outcome_assigned*/
                        when Outcome_Assigned&i. in ("LBM") And b.outcome in ("LBM" "UDL") and put(dxprrx ,$outcdpr.)='1' then Enc_Date
                        when Outcome_Assigned&i. in ("LBS") And b.outcome in ("LBS" "UDL") and put(dxprrx ,$outcdpr.)='1' then Enc_Date
                        when Outcome_Assigned&i. in ("MLS") And b.outcome in ("MLS" "UDL") and put(dxprrx ,$outcdpr.)='1' then Enc_Date
-                       when Outcome_Assigned&i. in ("SB") And b.outcome in ("SB" "UDL") and put(dxprrx ,$outcdpr.)='1' then Enc_Date
+                       when Outcome_Assigned&i. in ("SB")  And b.outcome in ("SB" "UDL") and put(dxprrx ,$outcdpr.)='1' then Enc_Date
                        when Outcome_Assigned&i. in ("UDL") And b.outcome in ( "UDL") and put(dxprrx ,$outcdpr.)='1' then Enc_Date
+
+					   /*CDL: MODIFICATION -- Prioritize non-UDL delivery pr codes over UDL dx codes -- DISCUSS*/
+					   /*NOTE: Some UDLs dont have any UDL pr codes because they had discordant delivery codes. Deal with that here.
+                       If assigned outcome is UDL then pick earliest delivery proc code in outcome group*/
+                       when Outcome_Assigned&i. in ("UDL") And put(b.outcome,$outclass.)='Delivery' and put(dxprrx ,$outcdpr.)='1' then Enc_Date

```

Comment:
Prioritizing non-UDL pr codes over UDL dx codes in those cases where UDL assigned. Important because some UDLs may not have any UDL codes but have other delivery procedure codes.